### PR TITLE
Terminal with running code capabilities

### PR DESCRIPTION
**Pull request by calgagi**
- Three commands: run, restart, save
   * `run` attaches a single command to list of moves and sends POST request to backend
   * `restart` clears move/command list (called gameHistory in code)
   * `save` saves file by sending save request to backend. Also sets filename to be specific file

There is also `help` which lists out all of these commands (no descriptions yet).

### CODE REVIEW

**Comment by calgagi** [comment]
Also this PR includes all changes in #3

File: src/Run/Run.tsx [hunk 1]
```diff
@@ -86,32 +83,77 @@ class SpielServerRequest {
...
-            </Modal.Footer>
-        </Modal>
+        <Terminal
+            color='white'
+            backgroundColor='black'
+            barColor='black'
+            style={{ fontSize: "1.1em" }}
+            showActions={false}
+            commands={{
+                restart: {
```
Comment (montymxb) [inline src/Run/Run.tsx]: Sweet dude!

File: src/Run/Run.tsx [hunk 2]
```diff
@@ -86,32 +83,77 @@ class SpielServerRequest {
...
+    let [gameHistory, setGameHistory] = React.useState(Array<string>());
+
+    code = props.code;
+
+    function save(args, print) {
+        if (args._.length != 1) {
+            print("Error: Save expects only one argument: the file to be saved.");
+        } else {
+            filename = args._[0];
+            SpielServerRequest.save(filename, code)
```
Comment (montymxb) [inline src/Run/Run.tsx]: Code works now?

File: src/Run/Run.tsx [hunk 3]
```diff
@@ -86,32 +83,77 @@ class SpielServerRequest {
...
 const Run = (props) => {
-    let [input,setInput] = React.useState('');
 
-    function run() {
-        props.commands.push(input);
-        props.runCode();
-        setInput("");
+    let [gameHistory, setGameHistory] = React.useState(Array<string>());
+
+    code = props.code;
```
Comment (montymxb) [inline src/Run/Run.tsx]: Ohhh...maybe this line fixed it?

Comment (calgagi) [reply src/Run/Run.tsx]: Yeah I think this is the line that fixed it! Thanks :)

**Comment by calgagi** [comment]
Ok, that should be the last commit for tonight. Unless we can think of anything else

**Pull request merged**